lib/MusicBrainz/Server/Controller/Role/EditRelationships.pm

```diff
@@ -122,8 +122,11 @@ role {
                 $url_changes = $self->edit_relationships($c, $form, \@urls, $source);
             }
 
-            my @rels = grep { !$_->is_empty } $form->field('rel')->fields;
-            my $rel_changes = $self->edit_relationships($c, $form, \@rels, $source);
+            my $rel_changes = 0;
+            if (defined $source) {
+                my @rels = grep { !$_->is_empty } $form->field('rel')->fields;
+                $rel_changes = $self->edit_relationships($c, $form, \@rels, $source);
+            }
```

Comment:
$source is not defined after the URL has been merged away. This change seems to work, but if there's a better option I am missing, do let me know! :)

Reply:
What if both the URL change + relationship changes are submitted? Are they not propagated to the merged URL?

Reply:
Hmm, true, I tested now and that would drop that rel change. Sigh. Thanks for catching it. Seems like this only works right now because we don't auto-apply any merges, so the relationship changes are entered on time... Looking into it.

Another thing that seems weird is that these are not run in a transaction or anything - you can have the edit changing the URL pass, then an ISE happen when trying to change rels, and the URL change will still have applied.

Reply:
I can get this to work by adding this inside `if ($edit)` above (updating the PR since it seems like an improvement):

```
                # If a URL got merged away, $source will be undefined now,
                # so we should follow the gid redirect to find the new entity
                # whose rels should change.
                if ($form_name eq "edit-url" && !$source) {
                    my $old_url_gid = $edit->{data}->{entity}->{gid};
                    $source = $c->model('URL')->get_by_gid($old_url_gid);
                }
```

But that seems to still cause issues for this case: 

URL1 `http://example.com` has "picture for: X"
URL2 `https://example.com` has "picture for: X"
URL1 is edited so that the URL is now `https://example.com` and the rel is now "homepage for: X"
The merge goes through first, but then the "picture for: X" rel that was being changed got merged with that of URL2. As such, the change to make it "homepage for: X" is silently dropped, because the rel it was meant to change is now gone.
I guess this is only a problem for URLs because no other entities can be merged *and* have their rels edited at the same time.
What should we do with that case?

Reply:
What if we processed the rel edits on the source before doing the merge? Seems like the merge should do the correct thing in that case if they're all submitted as auto-edits at least.